File: backend/src/Blizka.App/Domain/Entities/Idea.cs
using Blizka.App.Domain.Enums;

namespace Blizka.App.Domain.Entities;

public sealed class Idea
{
    public Guid Id { get; set; }

    public Guid AuthorUserId { get; set; }

    public User? AuthorUser { get; set; }

    public string Text { get; set; } = string.Empty;

    public bool IsAnonymous { get; set; }

    public IdeaStatus Status { get; set; } = IdeaStatus.New;

    public int VotesCount { get; set; }

    public DateTimeOffset CreatedAt { get; set; }

    public ICollection<IdeaVote> Votes { get; set; } = [];
}

File: frontend/src/domains/ideas/types/idea.ts
/**
 * Статус идеи на доске (S-60). Сверено с backend:
 * `Blizka.App/Domain/Enums/IdeaStatus.cs` — энумы сериализуются camelCase.
 */
export type IdeaStatus = 'new' | 'underReview' | 'planned' | 'implemented' | 'declined'

/** Статусы, которые на доске читаются как «этим занимаются». */
export const IN_PROGRESS_STATUSES: readonly IdeaStatus[] = ['underReview', 'planned']

/**
 * Идея с доски. Собрано по `Blizka.App/Domain/Entities/Idea.cs` и разделу
 * T-19.1 `decomposition.md` — самого эндпоинта ещё нет, см. docs/api-gaps.md.
 */
export type Idea = {
  id: string
  text: string
  status: IdeaStatus
  votesCount: number
  /** Голосовал ли текущий пользователь — от этого зависит вид кнопки. */
  hasVoted: boolean
  /** `null` — автор скрыл имя (`isAnonymous`). */
  authorName: string | null
  /** Идею предложил текущий пользователь — она попадает во вкладку «Мои». */
  isMine: boolean
  createdAt: string
}

/** Вкладки доски (S-60). Первые две — это `?sort=`, вторые две — фильтры. */
export type IdeaTab = 'hot' | 'new' | 'inProgress' | 'mine'
